### ISSUE TITLE
tzlocal uses startup timezone; parse uses runtime timezone

### ISSUE THREAD

**Issue by mjschultz** [dateutil/dateutil#100]
There appears to be a bug in the library depending on the import ordering and other libraries calling [`tzset`](https://github.com/django/django/blob/7da3923ba0c569aa23d0ab0a47a124af60a18f5b/django/conf/__init__.py#L133-134). Below is a snippet that demonstrates the bug without extra libraries.

When the `tzlocal` class is created it uses the current [`time.timezone`/`time.altzone`](https://github.com/dateutil/dateutil/blob/master/dateutil/tz.py#L108) to find a `_dst_offset` which is used at runtime. This class variable should remain constant through the life of the process.

Over in the parser however, the runtime value of [`time.tzname`](https://github.com/dateutil/dateutil/blob/master/dateutil/parser.py#L562-563) is checked and if it matches the parsed offset, a timezone of `tzlocal` is given.

Of course, if something has modified the process timezone via a `tzset` we now have a situation where we've classified something as `tzlocal()` even though the offset of that `tzlocal()` does not match the expected offset.

(And of course, since UTC never has daylight savings time I end up with the UTC-6 offset!)

I am currently in US/Central (UTC-5, Central Daylight Time).

```python
>>> from dateutil.tz import tzlocal
>>> from dateutil.parser import parse
>>> from datetime import datetime
>>> import time
>>> time.tzname
('CST', 'CDT')
>>> from os import environ
>>> environ['TZ'] = 'UTC'
>>> time.tzset()
>>> time.tzname
('UTC', 'UTC')
>>> t = parse('2015-07-20T20:52:19+00:00')
>>> t
datetime.datetime(2015, 7, 20, 20, 52, 19, tzinfo=tzlocal())
>>> print t
2015-07-20 20:52:19-06:00
>>> tzlocal().utcoffset(datetime(2015, 7, 20, 20, 52, 19))
datetime.timedelta(-1, 64800)
```

**Comment by mjschultz** [comment]
I've made a PR in https://github.com/dateutil/dateutil/pull/101, comments are welcomed. This was the simplest method I could think of that wouldn't have performance ramifications.

**Comment by pganssle** [comment]
Fixed by #107 and #109
